cmd/bot/main.go

```diff
@@ -149,7 +167,102 @@ func main() {
 	}
 }
 
+func isMessage(update map[string]any) bool {
+	_, hasMessage := update["message"]
+	return hasMessage
+}
+
+func isCallbackQuery(update map[string]any) bool {
+	_, hasCallbackQuery := update["callback_query"]
+	return hasCallbackQuery
+}
+
+func userID(update map[string]any) (uint64, error) {
+	var typ string
+
+	switch {
+	case isMessage(update):
+		typ = "message"
+	case isCallbackQuery(update):
+		typ = "callback_query"
+	default:
+		return 0, errUnknownUpdateType
+	}
+
+	return uint64(update[typ].(map[string]any)["from"].(map[string]any)["id"].(float64)), nil
+}
+
+func unknownCommand(log *slog.Logger, userID uint64) error {
+	_, err := doTelegramRequest(log, "sendMessage", url.Values{
+		"chat_id": {strconv.FormatUint(userID, 10)},
+		"text":    {"Sorry, ik begrijp je niet. Probeer /start of /privacy."},
+	})
+	return err
+}
+
+func privacy(log *slog.Logger, userID uint64) error {
+	_, err := doTelegramRequest(log, "sendMessage", url.Values{
+		"chat_id": {strconv.FormatUint(userID, 10)},
+		// This way we can use Markdown code formatting in a raw string literal.
+		"text":         {strings.ReplaceAll(fmt.Sprintf(privacyPolicy, userID), "~", "`")},
+		"parse_mode":   {"Markdown"},
+		"reply_markup": {privacyGotItMarkup},
+	})
+
+	return err
+}
+
+func handleCommand(log *slog.Logger, userID uint64, text string) error {
+	userIDString := strconv.FormatUint(userID, 10)
+
+	switch text {
+	case "/start":
+		if _, err := doTelegramRequest(log, "sendMessage", url.Values{
+			"chat_id":      {userIDString},
+			"text":         {"Hallo! In privé-chats kan ik niet zo veel. Mijn kanaal @energieprijzen is veel interessanter."},
+			"reply_markup": {startReplyMarkup},
+		}); err != nil {
+			return err
+		}
```

Comment:
Maybe this should be factored out into its own function for consistency with the other cases.

Or we could even do something like:

```go
command, ok := map[string]any{ // TODO: Figure out the actual type.
	"/start":   start,
	"/privacy": privacy,
}[text]

command(log, userID)
```